File: lv2/이진 변환 반복하기/code.cpp
#include <string>
#include <vector>

using namespace std;
int zeroCnt = 0;
int removeResult(string& str){
    int cnt1 = 0;
    for(char& c : str){
        if(c == '0'){
            zeroCnt++;
            continue;
        }
        cnt1++;
    }
    return cnt1;
}
string intToBinary(int length){
    string binaryStr = "";
    while(length != 0){
        binaryStr = to_string(length%2) + binaryStr;
        length /= 2;
        
    }
    return binaryStr;
}

vector<int> solution(string s) {
    vector<int> answer;
    string str = s;
    int funcCnt = 0;
    int length = 0;
    while(true) {
        length = removeResult(str);
        str = intToBinary(length);
        funcCnt++;
        if(str == "1")
            break;
    }
    answer.push_back(funcCnt);
    answer.push_back(zeroCnt);
    return answer;
}

File: lv1/푸드 파이트 대회/code.java
class Solution {
    public String solution(int[] food) {
        StringBuilder sb = new StringBuilder();
        sb.append("0");
        for(int i = food.length-1; i > 0; i--){
            for(int j = 0; j < food[i]/2; j++){
                sb.insert(0, i);
                sb.append(i);
            }
        }
        return sb.toString();
    }
}
